secret/secrets_manager_vault.go

```diff
@@ -192,10 +192,22 @@ func (m *BasicSecretsManager) DeleteSecret(ctx context.Context, id string) error
 		ForceDeleteWithoutRecovery: aws.Bool(true),
 		SecretId:                   &id,
 	})
-	return err
+	if err != nil {
+		return err
+	}
+
+	if !m.usesCache() {
+		return nil
+	}
+
+	if err := m.cache.Delete(ctx, id); err != nil {
```

Comment:
Can just `return errors.Wrapf(err, "deleting secret '%s' from cache", id)` here without any if statement

Reply:
Yeah, we can. I just have a slight personal preference to doing using `if err` blocks, because oftentimes, I find that I end up re-adding the `if err` statement anyways when I want to do anything after the last operation in the function body.

Reply:
Ok